lib/linked_list.rb

```diff
@@ -120,22 +199,46 @@ def has_cycle
     # Time Complexity:  
     # Space Complexity
     def get_first
-      raise NotImplementedError
+      if @head
+        return @head.data
+      else
+        return nil
+      end
     end
 
     # method that inserts a given value as a new last node in the linked list
     # Time Complexity:  
     # Space Complexity
     def add_last(value)
-      raise NotImplementedError
+    
+      new_node = Node.new(value, nil)
+      current = @head
+      if @head == nil
+        @head = new_node
+        return
+      else 
+        while current != nil

```

Comment:
Could you maybe make this loop  `until current.next.nil?`

Would that simplify it?